dynamic metrics.md

```diff
@@ -0,0 +1,165 @@
+
+* Good: No new columns. Keeps the "one metric per row" structure of the tibble. Easily unnested. Some existing precedent in lists of tibbles in tidymodels (e.g. `multi_predict()`)
+* Bad: Different column types for different metrics. 
+
+### Proposed Solution
+
+The third option is the proposed solution with the caveat that only metrics associated with the censored regression mode.
+
+For example, the standard structure for classification and regression results: 
+
+
+
+```r
+library(tidymodels)
+
+tidymodels_prefer()
+options(pillar.advice = FALSE)
+
+reg_metrics <- metric_set(rmse, rsq, ccc)
+reg_metrics(solubility_test, truth = solubility, estimate = prediction)
+```
+
+```
+## # A tibble: 3 × 3
+##   .metric .estimator .estimate
+##   <chr>   <chr>          <dbl>
+## 1 rmse    standard       0.722
+## 2 rsq     standard       0.879
+## 3 ccc     standard       0.937
+```
+
+As a mock-up of the new results, suppose that a static survival metric is used (`"surv_concordance"`) with a dynamic metric ("`surv_brier"`) used at time points 10, 20, and 30. The _metric set_ results would be: 
+
+
+```
+## # A tibble: 2 × 3
+##   .metric          .estimator .estimate       
+##   <chr>            <chr>      <list>          
+## 1 surv_concordance standard   <tibble [1 × 2]>
+## 2 surv_brier       standard   <tibble [3 × 2]>
+```
+
+The tibble corresponding to the row for concordance has a missing time point: 
+
+
+```
+## # A tibble: 1 × 2
+##   .time .estimate
+##   <lgl>     <dbl>
+## 1 NA          0.5
+```
+
+and the Brier score tibble has three rows: 
+
+
+```
+## # A tibble: 3 × 2
+##   .time .estimate
+##   <dbl>     <dbl>
+## 1    10     0.974
+## 2    20     0.164
+## 3    30     0.255
+```
+
+The list column of tibbles would only occur when the model mode is "censored regression". Since a metric set cannot mix metrics from different modes, the column will be type stable (conditional on the mode). 
+
+### Scope
+
+There very well might be other situations in the future where there are additional parameters (similar to `.time`). We believe that a list column approach may be used to address these issues. 
+
+Two potential uses are: 
+
+* For some spatial metrics, additional _data_ must be attached to compute additional metrics. The documentation for the [waywiser](https://github.com/mikemahoney218/waywiser) package shows an example where spatial geometry data is returned  to calculate spatial weights.
+
+* There is a tidymodels proposal to compute model characterization values to optionally return with performance metrics. For example, for a simple decision tree, we might return the number of terminal nodes, the number of predictors used in the splits, and the names of these predictors. While the first two quantities are scalars, the last is a vector of predictor names.
+
+It is possible that the approach listed above could accommodate situations such as these. 
```

Comment:
I don't immediately understand how this would extend to work with spatial data. In that situation, you need the actual geometry for each row, which is an (effectively) unique identifier. You also need _all_ of these values (for the assessment set, at least) at the same time; the metrics depend on the relative positions of each point in the assessment set to each other, not just the position of the current point.

That said, I've started thinking that the actual workflow in practice for spatial metrics isn't going to be "look at the mean spatial metrics for each assessment set in fit_resamples", but rather "connect the predictions from fit_resamples back to the original data and estimate the spatial metrics once, for the entire data set, using hold-out predictions". That's not going to require any changes in yardstick, I _think_, and so I think it's fine to do the optimal thing for survival analyses without worrying about spatial data.

Reply:
> I've started thinking that the actual workflow in practice for spatial metrics...

For context: no one is really tuning models based on spatial error distributions (for the same reason you don't tune based on bias, as it doesn't improve accuracy); they're checking spatial autocorrelation to either justify ignoring spatial structure in their model or to identify "problem areas" and potential ways to mitigate them. I think a more common workflow is where parameters are tuned using `rmse()`, and then spatial autocorrelation is checked before the model is "finalized".